userland/Makefile

```diff
@@ -1,20 +1,28 @@
 # userland master makefile. Included by application makefiles
 
-TOCK_USERLAND_BASE_DIR ?= $(abspath .)
-TOCK_BASE_DIR ?= $(abspath ../)
-BUILDDIR ?= $(abspath .)
+TOCK_USERLAND_BASE_DIR ?= ..
+TOCK_BASE_DIR ?= $(TOCK_USERLAND_BASE_DIR)/..
+BUILDDIR ?= build/$(TOCK_ARCH)
 TOCK_BOARD ?= storm
 TOCK_ARCH ?= cortex-m4
 LIBTOCK ?= $(TOCK_USERLAND_BASE_DIR)/libtock/build/$(TOCK_ARCH)/libtock.a
 
 TOOLCHAIN := arm-none-eabi
 
 # This could be replaced with an installed version of `elf2tbf`
-ELF2TBF ?= cargo run --manifest-path $(TOCK_USERLAND_BASE_DIR)/tools/elf2tbf/Cargo.toml --
+ELF2TBF ?= cargo run --manifest-path $(abspath $(TOCK_USERLAND_BASE_DIR))/tools/elf2tbf/Cargo.toml --
 ifdef PKG_NAME
 ELF2TBF_ARGS = -n $(PKG_NAME)
 endif
 
+# Files to compile. Only what is in the app directory.
```

Comment:
Hrm... Does this setup let us fine tune which c files are actually compiled in an app specific makefile? E.g. what if the sources are in a subdirectory or in multiple dirs? What if there are c files we want to exclude?